<a href="https://colab.research.google.com/github/lcanoi/MachineLearning_NoFrameworks/blob/main/Analysis_MR_M2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Momento de Retroalimentación: Módulo 2 Implementación de una técnica de aprendizaje máquina sin el uso de un framework.


In [ ]:
from google.colab import drive

drive.mount("/content/gdrive")  
!pwd  # show current path 

Mounted at /content/gdrive
/content


In [ ]:
%cd "/content/gdrive/MyDrive/Colab Notebooks/Data1"
!ls  # show current directory

/content/gdrive/MyDrive/Colab Notebooks/Data1
adult.data  iris.data  iris.names  wine.data  wine.names


In [ ]:
import pandas as pd

wineColumns = ["output",'f1','f2','f3','f4','f5','f6','f7','f8','f9','f10','f11','f12','f13']
df = pd.read_csv('wine.data', names=wineColumns)

input = df[['f1','f2','f3','f4','f5','f6','f7','f8','f9','f10','f11','f12','f13']]
output = df['output']

print('In:\n', input.head())
print(input.shape)
print('Out: \n', output.head())
print(output.shape)

In:
       f1    f2    f3    f4   f5    f6    f7    f8    f9   f10   f11   f12  \
0  14.23  1.71  2.43  15.6  127  2.80  3.06  0.28  2.29  5.64  1.04  3.92   
1  13.20  1.78  2.14  11.2  100  2.65  2.76  0.26  1.28  4.38  1.05  3.40   
2  13.16  2.36  2.67  18.6  101  2.80  3.24  0.30  2.81  5.68  1.03  3.17   
3  14.37  1.95  2.50  16.8  113  3.85  3.49  0.24  2.18  7.80  0.86  3.45   
4  13.24  2.59  2.87  21.0  118  2.80  2.69  0.39  1.82  4.32  1.04  2.93   

    f13  
0  1065  
1  1050  
2  1185  
3  1480  
4   735  
(178, 13)
Out: 
 0    1
1    1
2    1
3    1
4    1
Name: output, dtype: int64
(178,)


El número de entradas y salidas es fijo </br>
En cada instancia de vino contamos con 13 entradas (f1-f13) y una salida (output) </br>
Y contamos con 178 líneas de vinos

In [ ]:
df.isnull().sum()

output    0
f1        0
f2        0
f3        0
f4        0
f5        0
f6        0
f7        0
f8        0
f9        0
f10       0
f11       0
f12       0
f13       0
dtype: int64

Todos los datos son exactos y no hay valores nulos, por lo cual no es necesario hacer una limpieza adicional a la base de datos

Ahora nos falta determinar


1.   Número de Layers y Neuronas
2.   Pesos de la Red Neuronal
3.   Función de Activación
4.   Función de Costo



Para la Función de Activación comenzaremos probando con sigmoid

Siendo un problema de clasificación de vinos, para la Función de Costo se realizará Log-loss, informándonos de los outputs asignados a una clasificación correcta

In [ ]:
len(input.columns)

13

Contamos con 13 diferentes inputs, así que en nuestra capa de entrada contaremos con 13 neuronas

In [ ]:
output.unique()

array([1, 2, 3])

Existen 3 posibles clasificaciones para el vino (clasificación multi-clase), por lo cual en nuestra capa de salida contaremos con 3 neuronas